content/doc/book/security/credentials.adoc

```diff
@@ -0,0 +1,41 @@
+---
+title: Credentials
+layout: section
+---
+ifdef::backend-html5[]
+:toc:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+endif::[]
+
+Use credentials to secure access to
+external sites and applications that can interact with Jenkins such as artifact repositories, cloud-based storage systems and services, and databases.
+This is both more secure and more convenient that coding username and password or other authentication devices in each Pipeline.
+
+== Working with Credentials
+
+Implementing credentials has two elements:
+
+* Configure the credential. See
+link:/doc/book/using/using-credentials/#configuring-credentials[Configuring Credentials]
+on the
+link:/doc/book/using/using-credentials/[Using Credentials]
+page.
+
+* Call credentials in the Pipeline to gain access to an external resource.
+See
+link:/doc/book/pipeline/jenkinsfile/#handling-credentials[Handling Credentials
+on the
+link:/doc/book/pipeline/jenkinsfile/[Using a Jenkinsfile]
+page.
+
+== Best Practices for Credentials
+
+Need some good advice added here.
+I have listed some possible topics as a start but feel free to override.
+
+* Limit number of people with *Credentials > Create* permissions.
+* Modify the underlying credentials (username/password, secret text, etc) frequently.
+* Restrict access to each credential to the project, item, or user as appropriate.
```

Comment:
You can rephrase this by saying you want to limit the scope of who has access to the credentials. The objective is to restrict to the minimum the attack surface.

As we discussed the topic during our talk at DevOps World 2019 ["Thinking about Security"](https://drive.google.com/file/d/1RisjNUfu-3_VOyTIvY0cdWWzM8HKDtY9) (slide 24), it's easier to build from the bottom (no grant) and add people instead of from the top (giving to everyone) and then removing grants.

Reply:
Thanks, Waddeck!  I rewrote this section but I  may not yet have it right.  We now have a subsection about restricting access, which allows a more readable discussion of the issues and such.  Please critque ruthlessly!

I also added a subsection about protecting secrets.